barcodes_generator_abstract/models/barcode_generate_mixin.py

```diff
@@ -29,14 +29,25 @@ class BarcodeGenerateMixin(models.AbstractModel):
         related="barcode_rule_id.generate_type",
     )
 
+    def write(self, vals):
+        res = super(BarcodeGenerateMixin, self).write(vals)
+        if vals.get("barcode_rule_id"):
+            rule = self.env["barcode.rule"].browse(vals["barcode_rule_id"])
+            if rule.generate_automate and rule.generate_type == "sequence":
+                for rec in self:
+                    if not rec.barcode:
+                        rec.generate_base()
+                        rec.generate_barcode()
```

Comment:
```suggestion
                    if not rec.barcode_base:
                        rec.generate_base()
                     if not rec.barcode:
                        rec.generate_barcode()
```

(if base is defined, regenerate it is not required, don't you think ?)